src/patterns/passwords/index.md

```diff
@@ -69,6 +69,12 @@ To help users meet your password constraints and prevent mistyped passwords, you
 - show the last typed character of their password
 - make them enter their password twice and automatically compare them
 
+#### Showing and hiding passwords
+One common method for helping users type passwords is a show/hide button near the password input. When the password is hidden, the button will change the input styling to 'show' the password. When the password is visible, the button will 'hide' the password again.
+
+When there are two or more password fields on a page, the 'show' and 'hide' labels for each password input must be different.
+For example, you can label the input "Password" as "show first password" and label the second input "Re-enter password" as "show second re-entered password".
```

Comment:
Thanks @davidc-gds, @claireashworth 

The guidance on the 2nd half sounds good. I'm just wondering what's the use case in this instance?

Is the user entering 2 separate passwords for some reason (which the example suggests and is a bit confusing)...
...or is this an 'enter password' and 'confirm password' (so labels would be 'show password' and 'show password to confirm'?)?

Providing the contextual info (1st half), I'm wondering if linking to backlog items, or other links, would be a less committal way to provide the context we need. Otherwise it feels like we're going to end up explaining and look to be endorsing a lot of non-DS components and approaches.

Like how about:

```suggestion
It's common for services to help users type passwords with a '[show password](https://github.com/alphagov/govuk-design-system-backlog/issues/240)' button to let users see their password if they want to.

When there are two or more password fields on a page, the 'show' and 'hide' labels for each password input must be different.

For example, you can label the input "Password" as "show password" and label the second input "Re-enter password" as "show re-entered password".
```

Reply:
Hey Calvin! The use case for this is going to mostly be for account creation pages, where the user is selecting their password. So one input will be the password, and then the second input will be confirming the password.

The only other common scenario I can think of, which involves 3 password inputs, is changing a password. Then you would have inputs that labelled something like 'current password', 'new password' and 'confirm new password'.

Reply:
Okay, this looks okay to me, as a thought exercise here's what the inputs and show/hide labels would be:
- Password [show password]
- Re-enter password [show re-entered password]
- Current password [show current password]
- New password [show new password]
- Re-enter new password [show re-entered new password]